create_derivatives/routines.py

```diff
@@ -65,8 +67,103 @@ def get_as_uri(self, bag_filepath):
 
 
 class JP2Maker:
-    # TO DO: make JPG2000 derivatives
-    pass
+    """Creates JP2000 derivatives from TIFFs.
+
+    Creates JP2 directory in bag's data directory and JP2 derivatives.
+    TIFFs to be converted are targeted in the service directory. If the service directory is empty or
+    does not exist, target TIFFs at the root of the data directory.
+
+    Returns:
+        A tuple containing human-readable message along with list of bag identifiers.
+        Exceptions are raised for errors along the way.
+    """
+
+    def run(self):
+        bags_with_jp2s = []
+        for bag in Bag.objects.filter(process_status=Bag.PREPARED):
+            service_dir = Path(bag.bag_path, "data", "service")
+            if service_dir.is_dir() and len(service_dir):
+                tiff_files_dir = str(Path(bag.bag_path, "data", "service"))
+            else:
+                tiff_files_dir = str(Path(bag.bag_path, "data"))
```

Comment:
I would suggest coercing to a string where this variable is used below (`create_jp2`, when calling `matching_files`). I am going to update that helper so it takes a pathlib object rather than a string, which I think will reduce a lot of this converting/to from strings throughout the app.